src/modsim2/attack/adversarial_images.py

```diff
@@ -101,9 +101,54 @@ def generate_adversarial_images(
     # Put model into foolbox format
     fmodel = fb.PyTorchModel(model, bounds=(0, 1), device=device)
 
-    # Generate attack images
-    attack = getattr(fb.attacks, attack_fn_name)(**kwargs)
-    _, clipped_advs, success = attack(fmodel, images, labels, epsilons=epsilons)
+    # If Boundary Attack, need to initialise and manage early failures to avoid error
+    if attack_fn_name == "BoundaryAttack":
```

Comment:
large `if`, probably worth wrapping up in a fn for clarity

Reply:
done